### ISSUE TITLE
Some Issues I Found.

### ISSUE THREAD

**Issue by mifth** [GuilhermeGSousa/game-base-2D#3]
Hi, I found some issues according to Camera scripts. I recorded a video for you.
[2023-10-31 21-46-01.zip](https://github.com/GuilhermeGSousa/game-base-2D/files/13221271/2023-10-31.21-46-01.zip)

- camera_targeter.gd must use _physics_process()
- camera_shaker.gd must use _physics_process()
- _update_camera_zoom() function probably has some problems.

**Comment by GuilhermeGSousa** [comment]
Thank you for opening these issues I'll start looking at them today! I think you're right about the first 2 points, as for `_update_camera_zoom()` its effect won't be visible if you're only targeting one object. What it does is that it adapts the zoom to fit all of the targets, clamped to zoom min and max. So it needs to be updated every frame, that said it probably needs to be changed to avoid flickering, I'll look into that.

**Comment by GuilhermeGSousa** [comment]
The goal of the zoom update is to have something like this.
![zoom](https://github.com/GuilhermeGSousa/game-base-2D/assets/13313321/38096551-41f4-48c2-abbb-365e86fcef28)
I'm turning the `CameraTargeter` into a `@tool` so the update happens in editor, should make its behavior more obvious.

**Comment by mifth** [comment]
It's a nice feature. But I think most projects use a simple camera. I would not complicate so much a camera for starters.

**Comment by GuilhermeGSousa** [comment]
I don't agree with you there, its very useful if you want to be able to show something to the player, like always showing the boss during the boss fight. Having a camera that follows a character wouldn't require a script, you could just parent the camera to a node.

**Comment by mifth** [comment]
I totally agree that it's a nice feature. But I wouldn't put it into a starter project. KISS! Keep it simple. :)
